### ISSUE TITLE
[Suggestion] Redefine fonts for Polish and Russian

### ISSUE THREAD

**Issue by VladWinner** [fedddddd/h2-mod#301]
How about the client loads a good/proper font for Polish (`polish`) and Russian (`russian`&`russian_partial`) instead of a stylistically and metrically unsuitable fallback?

Comparison: https://imgur.com/a/yahuTZx
Notice how badly the letters in subtitles are rendered using the built-in fallback font.

These fonts are compatible with the following localizations/languages:
English, French, German, Italian, Polish, Portuguese, Russian, Spanish

Update: 
-----
I noticed that only two fonts (`default.otf` & `defaultBold.otf`) are applied in fallback mode, however, in other localizations (where this mode is disabled) `bank.ttf` is also used. 
![image](https://user-images.githubusercontent.com/13555921/178993426-a6587f92-ba0f-427c-a5b0-a9332d51eefa.png)
If there is a way to bypass this 'fallback' mode for Polish and Russian, it will be possible to use 3 fonts at once:
#### Download: [fonts_polrus3.zip](https://github.com/fedddddd/h2-mod/files/9112044/fonts_polrus3.zip)

`default.otf` & `defaultBold.otf` are fonts that I previously suggested for H1-Mod in https://github.com/h1-mod/h1-mod/pull/179, 
`bank.ttf` is based on OFL recreation of Bank Gothic ([source](https://github.com/tphinney/science-gothic/tree/main/fonts/variable)), but normalized in height, with tabular figures.

**Comment by liveskai** [comment]
Does anyone know how to change the language?

**Comment by VladWinner** [comment]
> Does anyone know how to change the language?
https://github.com/fedddddd/h2-mod/issues/298#issuecomment-1183855439

**Comment by VladWinner** [comment]
Just another example of font usage:
(before/after)
![1b](https://user-images.githubusercontent.com/13555921/179086279-a0d5b28c-f265-4481-9808-daece96413a7.png)

**Comment by fedddddd** [comment]
While I don't dislike the idea, there are some issues with the font you suggested:
![image](https://user-images.githubusercontent.com/58637860/179101917-f68a4c18-30e7-4bc6-85e4-0f135fa015f1.png)
Chinese/Korean/Japanese fonts are not supported (not a big issue)
Button text is at an offset

**Comment by VladWinner** [comment]
> Chinese/Korean/Japanese fonts are not supported (not a big issue)

Yes, I noticed that too. Custom (user) fonts now take priority over substitution rule in language.lua. I guess that's why .notdef glyph is displayed instead of special fonts for CJK languages.
I this that loading one big font with JCK glyphs included, even if the language is not Asian, would not be very optimal solution.

> Button text is at an offset

This happens due to an internal rule for built-in fallback font and only in languages where `LUI.UIButtonText.IsOffsetedLanguage()` function works.
![image](https://user-images.githubusercontent.com/13555921/179118349-a480aa7b-74f5-4b16-ad1c-6fc1ea4218d1.png)

Here is the solution:
```
LUI.UIButtonText.IsOffsetedLanguage = function ()
	if Engine.GetCurrentLanguage() == 5 or 6 then
		return false
	end
end
```
![image](https://user-images.githubusercontent.com/13555921/179118693-0e6b5fbf-8195-46ac-9a83-42c7230d8c45.png)

**Comment by VladWinner** [comment]
Many thanks for the implementation!!!

> I noticed that only two fonts (`default.otf` & `defaultBold.otf`) are applied in fallback mode, however, in other localizations (where this mode is disabled) `bank.ttf` is also used. 
![image](https://user-images.githubusercontent.com/13555921/178993426-a6587f92-ba0f-427c-a5b0-a9332d51eefa.png)

Weird, `bank.ttf` is still ignored for headers and HUD ammo counter in Polish and Russian localizations (forcibly replaced by `defaultBold.otf`), so actually only 2 fonts are used.

**Comment by fedddddd** [comment]
@VladWinner fixed in c96c9ff265b6f5cfe8e72d6c982f20f34db78aa6

**Comment by VladWinner** [comment]
> @VladWinner fixed in [c96c9ff](https://github.com/fedddddd/h2-mod/commit/c96c9ff265b6f5cfe8e72d6c982f20f34db78aa6)

Thanks! 
I’m sorry to bore you again with these fonts, but I saw another leftover from fallback font and found its culprit: modified gray digits (zeros) of HUD ammo counter. They even forgot to apply it for 'Russian (Partial)'. Can this be bypassed somehow?
![image](https://user-images.githubusercontent.com/13555921/179302355-9d5a9150-2c50-438d-a6d2-fe705ee230da.png)

**Comment by fedddddd** [comment]
@VladWinner fixed in 4fa5dbe9300f8fe24dc50b666588d6847254b981
